crates/astria-composer/src/searcher/executor.rs

```diff
@@ -252,6 +360,30 @@ impl Executor {
         Ok(())
     }
 }
+
+#[derive(Debug, thiserror::Error)]
+enum ExecutionError {
+    #[error("failed to submit sequencer transaction")]
+    TransactionSubmissionFailed {
+        error: SequencerClientError,
```

Comment:
call this `source` (preferred) or add a `#[source]` attribute. See the thiserror docs